### PR TITLE
Let exclude_suites support regex

### PR DESCRIPTION
**Pull request by caesarxuchao**
It's particularly useful if we want to start the build while waiting for CI tests results.

### CODE REVIEW

**Comment by caesarxuchao** [comment]
@david-mcmahon PTAL. I forgot to assign it to you.

File: lib/releaselib.sh [hunk 1]
```diff
@@ -116,10 +116,22 @@ release::set_build_version () {
   local main_job="${all_jobs[0]}"
 
   # Loop through the remainder, excluding anything specified by --exclude_suites
-  for ((i=1;i<=${#all_jobs[*]};i++)); do
-    re="\\b${all_jobs[$i]}\\b"
-    [[ ${exclude_suites[*]} =~ $re ]] || secondary_jobs+=(${all_jobs[$i]})
+  for ((i=1;i<${#all_jobs[*]};i++)); do
+    match="false"
```
Comment (david-mcmahon) [inline lib/releaselib.sh]: Thanks for doing this!

Everything below here can be simplified to:
```
[[ -n $exclude_patterns && ${all_jobs[$i]} =~ $exclude_patterns ]] \
     || secondary_jobs+=(${all_jobs[$i]})
```

And then just define exclude_patterns just below exclude_suites above:
```
local -a exclude_suites=($3)
# below construct joins array
local exclude_patterns=$(IFS="|";echo "${exclude_suites[*]}")
```

Comment (caesarxuchao) [inline lib/releaselib.sh]: Done. Learned many bash tricks along the way ;)

File: lib/releaselib.sh [hunk 2]
```diff
@@ -116,10 +117,12 @@ release::set_build_version () {
...
   # Loop through the remainder, excluding anything specified by --exclude_suites
-  for ((i=1;i<=${#all_jobs[*]};i++)); do
-    re="\\b${all_jobs[$i]}\\b"
-    [[ ${exclude_suites[*]} =~ $re ]] || secondary_jobs+=(${all_jobs[$i]})
+  for ((i=1;i<${#all_jobs[*]};i++)); do
+    [[ -n $exclude_patterns && ${all_jobs[$i]} =~ $exclude_patterns ]] \
+        || secondary_jobs+=(${all_jobs[$i]})
   done
+  
+  logecho "After excluding $exclude_patterns, secondary_jobs are ${secondary_jobs[*]}"
```
Comment (david-mcmahon) [inline lib/releaselib.sh]: This list is generated as part of the normal output.  We don't need to display it here.

File: anago [hunk 3]
```diff
@@ -95,8 +95,8 @@ PROG=${0##*/}
 #+                                 source/include:
 #+                                 FLAGS_security_layer=/path/to/script
 #+                                 Default: $HOME/.kubernetes-releaserc
-#+     [--exclude-suites=]       - Space separated list of CI suites to exclude
-#+                                 from go/nogo criteria
+#+     [--exclude-suites=]       - Space separated list of CI suites regex to
```
Comment (david-mcmahon) [inline anago]: Can you update the man page in find_green_build too?

**Comment by caesarxuchao** [comment]
Comments addressed. PTAL. Thanks.

**Comment by caesarxuchao** [comment]
Thanks for the reviews.

**Pull request merged**